web/src/lottie-player.ts

```diff
@@ -0,0 +1,577 @@
+      data = JSON.parse(data);
+    } catch (err) {
+      const json = await _fromURL(data as string);
+      data = JSON.stringify(json);
+    }
+  }
+
+  const encoder = new TextEncoder();
+  return encoder.encode(data as string);
+}
+
+const _fromURL = async (url: string): Promise<Record<string, any>> => {
+  if (typeof url !== "string") {
+    throw new Error(`The url value must be a string`);
+  }
+
+  let json;
+
+  try {
+    // Try construct an absolute URL from the src URL
+    const srcUrl: URL = new URL(url);
+
+    // Fetch the JSON file from the URL
+    const result: any = await fetch(srcUrl.toString());
+
+    json = await result.json();
+  } catch (err) {
+    throw new Error(
+      `An error occurred while trying to load the Lottie file from URL`
+    );
+  }
+
+  return json;
+}
+
+const _wait = (timeToDelay: number) => {
+  return new Promise((resolve) => setTimeout(resolve, timeToDelay))
+};
+
+const _observerCallback = (entries: IntersectionObserverEntry[]) => {
+  const entry = entries[0];
+  const target = entry.target as LottiePlayer;
+  
+  if (entry.isIntersecting) {
+    if (target.currentState === PlayerState.Frozen) {
+      target.play();
+    }
+  } else if (target.currentState === PlayerState.Playing) {
+    target.freeze();
+    target.dispatchEvent(new CustomEvent(PlayerEvent.Freeze));
+  }
+}
+
+@customElement('lottie-player')
+export class LottiePlayer extends LitElement {
+  /**
+  * LottieFiles JSON data or URL to JSON.
+  * @since 1.0
+  */
+  @property({ type: String })
+  public src?: string;
+
+  /**
+   * Animation speed.
+   * @since 1.0
+   */
+  @property({ type: Number })
+  public speed: number = 1.0;
+
+  /**
+   * Autoplay animation on load.
+   * @since 1.0
+   */
+  @property({ type: Boolean })
+  public autoPlay: boolean = false;
+
+  /**
+   * Number of times to loop animation.
+   * @since 1.0
+   */
+  @property({ type: Number })
+  public count?: number;
+
+  /**
+   * Whether to loop animation.
+   * @since 1.0
+   */
+  @property({ type: Boolean })
+  public loop: boolean = false;
+
+  /**
+   * Direction of animation.
+   * @since 1.0
+   */
+  @property({ type: Number })
+  public direction: number = 1;
+
+  /**
+   * Play mode.
+   * @since 1.0
+   */
+  @property()
+  public mode: PlayMode = PlayMode.Normal;
+
+  /**
+   * Intermission
+   * @since 1.0
+   */
+  @property()
+  public intermission: number = 1;
+
+  /**
+   * total frame of current animation (readonly)
+   * @since 1.0
+   */
+  @property({ type: Number })
+  public totalFrame: number = 0;
+
+  /**
+   * current frame of current animation (readonly)
+   * @since 1.0
+   */
+  @property({ type: Number })
+  public currentFrame: number = 0;
+
+  /**
+   * Player state
+   * @since 1.0
+   */
+  @property({ type: Number })
+  public currentState: PlayerState = PlayerState.Loading;
+
+  private TVG?: any;
+  private canvas?: HTMLCanvasElement;
+  private imageData?: ImageData;
+  private beginTime: number = Date.now();
```

Comment:
private variables and methods have to have underscore

```suggestion
  private _TVG?: any;
  private _canvas?: HTMLCanvasElement;
  private _imageData?: ImageData;
  private _beginTime: number = Date.now();
```